### PR TITLE
feat(web): add link to external map in leaflet popup

### PR DESCRIPTION
**Pull request by lefuturiste**
A little life improvement.

Sometimes it's useful to open a geo location to an external map application to not have to copy the coordinates manually. Here I put a link to OpenStreetMap because it's what I personally use. But I known some people would want to use something different. We could instead link to geohacks (eg. https://geohack.toolforge.org/geohack.php?params=048.861085_N_0002.313158_E_globe:Earth) or make it a configurable param.

I'm eager to hear your takes on this.

image

### CODE REVIEW

File: web/src/lib/components/shared-components/leaflet/marker.svelte [hunk 1]
```diff
@@ -37,10 +37,14 @@
...
+      marker.bindPopup(popupHTML);
     } else {
       marker.unbindPopup();
     }
   }
 </script>
+
+<span contenteditable="true" bind:innerHTML={popupHTML} class="hide">
+    <slot/>
+</span>
```
Comment (lefuturiste) [inline web/src/lib/components/shared-components/leaflet/marker.svelte:50]: Here I used a little hack to capture the HTML content of the default slot passed to the Marker component, tell me if there is something more clean than that.

**Comment by jrasm91** [comment]
Can you add some screenshots or video to the PR description?

**Comment by lefuturiste** [comment]
> Can you add some screenshots or video to the PR description?

done

**Comment by lefuturiste** [comment]
> I'm not a huge fan of the marker with text personally. We could instead display the lat/long in the info panel along with the address and/or wrap the whole leaflet map with a link so clicking on it links to a map.

I'm not sure about the link on the whole map,  I think that the map should be at minimum interactive as it's the case right now. Being able to wander a little bit around, zoom out or zoom in to locate the place is important. I think the best is to have an explicit link with an explicit label, so the user known whats the purpose.

For putting the coordinates in the infos panel, why not, but where to put it? I don't think we have the room to put it on the same field with the locality (reversed geocoding result) .

**Comment by alextran1502** [comment]
I believe right now you can click on the icon to show the lat/lon right? Why don't you add a link to the pop up instead of always show it? It would make the UI cleaner and also serve your needs

**Comment by lefuturiste** [comment]
Yes, I didn't change the behavior, you still need to click on the marker for the popup to show up (perharps I should have recorded a video)

**Review by alextran1502 — APPROVED** [review]

File: web/src/lib/components/shared-components/leaflet/marker.svelte (hunk 1 above)
Comment (jrasm91) [reply web/src/lib/components/shared-components/leaflet/marker.svelte:50]: This is pretty clever and works for me :+1:

**Review by jrasm91 — APPROVED** [review]

**Pull request merged**